### DISCUSSION TITLE
Using Variables as Command Arguments

### DISCUSSION THREAD

**Discussion by vincent-herlemont** [nushell/nushell#11627] [Q&A]
Context:
- NuShell version: 0.88.1
- Operating System: Ubuntu 20.04.6 LTS

I am experiencing an issue when trying to use variables as command arguments in NuShell. Specifically, when I define a variable that contains multiple command line options and then attempt to use this variable with the `glob` command, I get a type mismatch error.

### Steps to Reproduce
1. Define a variable with multiple command options:
    ```nu
    let option_type = ['--no-dir' '--no-symlink']
    ```

2. Use the variable with the `glob` command:
    ```nu
    glob $option_type... **
    ```

3. The following error is observed:
    ```
    Error: nu::parser::type_mismatch
      × Type mismatch.
       ╭─[entry #4:1:1]
     1 │ glob $option_type... **
       ·      ───────┬───────
       ·             ╰── expected string, found range
       ╰────
    ```

I need some advices to achieve this kind of workflow: store options on a variable and use it and command arguments.

**Comment by fdncred** [comment]
Not all internal commands understand variables. Apparently, glob is one that doesn't.
